### DISCUSSION TITLE
Missing files and incompatible .Net version

### DISCUSSION THREAD

**Discussion by abcd3435** [AvaloniaUI/Avalonia#9123] [General]
The first error i get, is about some missing files, which should be here by default, since it's a new project
![img1](https://user-images.githubusercontent.com/98709007/194694083-24b52158-9bae-4273-b622-04947009354b.png)

The project is also missing almost all files, but they get created
![img4](https://user-images.githubusercontent.com/98709007/194694192-01a2b450-aea1-445f-8628-f68524e34549.png)

The next problem is about .Net 6 missing assemblies
![img2](https://user-images.githubusercontent.com/98709007/194694494-71eedec3-09b0-48f9-9e96-b4d3fb70c21e.png)
![img3](https://user-images.githubusercontent.com/98709007/194694230-fb8f61b8-10b9-4239-ad61-e1a4c7da4e32.png)

I am on Windows 8.1 pro and VS 2019, so the older versions should work but i can't find any, i tried to run a project on VS 2022, and still the situation was even worse, most likely because of very poor compatibility with win 8.1 and many other bugs.
I can't install .Net 6 for VS 2019, i will see if i can change the target version, but idk why the files are not added into the project, and some of them are missing.

I changed the target, and seems at least that error is gone, but still the project isn't usable, it's missing files for some reason.
I tried to add all the files manually, but it doesn't work since it says the files are duplicated.
![img5](https://user-images.githubusercontent.com/98709007/194695102-64ef620c-2ed6-44a2-8bbd-963932739276.png)

**Comment by timunie** [comment]
Have you updated Avalonia to version 11.0 preview? If so, make sure that the theme is now in it's own nuget package.

**Reply by abcd3435** [reply]
> Have you updated Avalonia to version 11.0 preview? If so, make sure that the theme is now in it's own nuget package.

I don't think so, what got installed was the default plugin from the marketplace, for vs studio 2017 and 2019, and that's it. From what i remember it was version 10 something. What i also found is that the default templates for vs studio somehow are missing, and i couldn't start a new console application project, i will reinstall everything and then see.

**Reply by timunie** [reply]
Sometimes VS doesn't install nuget properly, I don't know why. Maybe it's the issue

**Reply by abcd3435** [reply]
> Sometimes VS doesn't install nuget properly, I don't know why. Maybe it's the issue

Nothing, the templates didn't get fixed, and i belive it has something to do with avalonia not working, and it's the standard templates that don't work, i tried console app and winforms, but nothing, same error
![img0](https://user-images.githubusercontent.com/98709007/194706359-b63f65c1-e231-46c3-9d85-5a8740d27505.png)

I don't understand why is it looking for .Net ver 6... Also, the folder that it's looking for clearly doesn't exist. I tried to look up over the internet, but i coulden't find any real solution. I am not very familiar with vs studio, but probably changing the version from 6 to about 4.8 and specify the path for these templates might fix the issue, but i don't know where are these settings.

**Reply by abcd3435** [reply]
> Sometimes VS doesn't install nuget properly, I don't know why. Maybe it's the issue

I look it up and it says that vs should look for the highest version installed and set it as default, so i uninstall the version i got, added a global jsonfile with new version as they suggested, but still it's looking for version 6 when creating the project... Idk what is this, some nasty stuff.

![img1](https://user-images.githubusercontent.com/98709007/194710214-4a2b30ef-7d11-4935-bfe1-d579b92d307a.png)
If i list the current version, 5.0.408 is shown, but why it's still loking for 6 idk, any idea how to fix this?

**Comment by danwalmsley** [comment]
looks like: Visual studio has a bad install, run visual studio installer and install updates.

also from command line: `dotnet restore` from you project directory.

**Reply by abcd3435** [reply]
> looks like: Visual studio has a bad install, run visual studio installer and install updates.
> 
> also from command line: `dotnet restore` from you project directory.

Damn it, in around 7 years of hobbyist programming, i never got a fully working vs... That's remarkable achievement for MS, overblown buggy stuff. Unluckly jetbrains rider isn't really free, otherwise i would take it and never look back.

**Reply by abcd3435** [reply]
I downloaded again vs 2019 with less stuff,, now the templates do work. Still avlonia doesn't work, the project now loads correctly, the files are included, but now i can't set the target framework directly in project settings, nothing is available, had to figure it out and set it in the project file manually, also by default c# ver. 7.3 is selected, which also doesn't work and compile, so again i had to switch it manually in the project file.

If i compile a new project, everything works, but as soon as i add a button, then i get compiler errors, i literally copy the 3 lines of code from button example... Trully pathetic.

![img1](https://user-images.githubusercontent.com/98709007/194756866-9f172a54-7570-4613-b71a-2ff4b822cec2.png)
What to do with this error?

**Reply by timunie** [reply]
That's a different error. It tells you, that content can only be one single item. If you want multiple controls, add a panel around it. Something like Grid or StackPanel.

**Reply by timunie** [reply]
For basic samples see here https://github.com/AvaloniaUI/Avalonia.Samples

**Reply by abcd3435** [reply]
> That's a different error. It tells you, that content can only be one single item. If you want multiple controls, add a panel around it. Something like Grid or StackPanel.

I should have mention that i already did that, and same error orrurs, i copy the button example and add it to the default generated file.
<StackPanel>
<Button Width="160" >My Button</Button>
</StackPanel>

**Reply by timunie** [reply]
The error says you have two contents in one root. If you share your current state, I can double check what's wrong.

**Reply by rabbitism** [reply]
> The error says you have two contents in one root. If you share your current state, I can double check what's wrong.
Hmmm, putting two elements inside Window definitely will trigger this error.

**Reply by abcd3435** [reply]
The error says you have two contents in one root. If you share your current state, I can double check what's wrong.

Sure, but i already mention, that it's literally the copy of the example, since the the button inside <StackPanel> didn't work, i posted the version without it, but here it is:

<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication4.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication4.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication4">
	
	<StackPanel>
		<Button Width="160">My Button</Button>
	</StackPanel>
	
	<Design.DataContext>
        <vm:MainWindowViewModel/>
    </Design.DataContext>
    <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center"/>
</Window>

(I upload a screenshot, since for some reason the code isn't fully shown)
![img1](https://user-images.githubusercontent.com/98709007/194924490-d3dafbe8-bb79-41a1-bc4b-8c096b7b2247.png)

The example is here: https://docs.avaloniaui.net/docs/controls/buttons/button
I don't understand why it doesn't work, it's literally the copy of the example, i hope it's not some kind of another unknown bug present in my vs studio.

**Reply by timunie** [reply]
You still have two contents in your Window: 

```xml
<!-- First child, OK -->
<StackPanel>
	<Button Width="160">My Button</Button>
</StackPanel>

<Design.DataContext>
    <vm:MainWindowViewModel/>
</Design.DataContext>

<!-- Second child, Not OK -->
<TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center"/>
```

Now, if you reorder like this it should work:

```xml
<Design.DataContext>
    <vm:MainWindowViewModel/>
</Design.DataContext>

<!-- Single child for Window, OK -->
<StackPanel>
        <!-- You can add several controls here -->
	<Button Width="160">My Button</Button>
        <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center"/>
</StackPanel>
```

**Reply by abcd3435** [reply]
> You still have two contents in your Window:
> 
> ```
> 
> <StackPanel>
> 	<Button Width="160">My Button</Button>
> </StackPanel>
> 
> <Design.DataContext>
>     <vm:MainWindowViewModel/>
> </Design.DataContext>
> 
> 
> <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center"/>
> ```
> 
> Now, if you reorder like this it should work:
> 
> ```
> <Design.DataContext>
>     <vm:MainWindowViewModel/>
> </Design.DataContext>
> 
> 
> <StackPanel>
>         
> 	<Button Width="160">My Button</Button>
>         <TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center"/>
> </StackPanel>
> ```
I didn't know that the textblock was a widget, it wasn't inside the stackpanel so i tought it was something else, but now it looks like it finnaly does work.
